File: seminar5/model/HotDrink.java
package org.example.hw.seminar5.model;

import java.util.Scanner;

public class HotDrink extends Product{

    private int temperature;
    public HotDrink(String name, int price, int volume, int temperature) {
        super.name = name;
        super.price = price;
        super.volume = volume;
        this.temperature = temperature;
    }

    public HotDrink() { }

    public int getTemperature() {
        return temperature;
    }

    public void setTemperature(int temperature) {
        this.temperature = temperature;
    }

    @Override
    public Product createProduct() {
        System.out.println("Ввод по очереди названия, цены, объема и температуры");
        Scanner scanner = new Scanner(System.in);
        String name = scanner.next();
        int price = enterNumber();
        int volume = enterNumber();
        int temperature = enterNumber();
        return new HotDrink(name, price, volume, temperature);
    }

    @Override
    public String toString() {
        return String.format("HotDrink//%s/temperature:%d",super.toString(), this.temperature);
    }
}

File: seminar7/Hot_drink.py
from Product import Product

class Hot_drink(Product):
    def __init__(self, name, price, volume, temperature):
        super().__init__(name, price, volume)
        self._temperature = temperature

    def __str__(self):
        return "{} {} {} {}".format(self._name, self._price, self._volume, self._temperature)

    @property
    def temperature(self):
        return self._temperature
    
    @temperature.setter
    def temperature(self, new_temperature):
        self._temperature = new_temperature